mass_mailing_newsletter_welcome_mail/data/base_automation_data.xml

```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo noupdate="1">
+    <data>
+        <record id="email_template" model="mail.template">
+            <field name="name">Welcome mail to newsletter subscribers</field>
+            <field name="model_id" ref="mass_mailing.model_mail_mass_mailing_contact"/>
+            <field name="use_default_to" eval="True"/>
+            <field name="lang">${object.env.context.get("lang")}</field>
+            <field name="subject">Welcome to the mailing list</field>
+            <field name="body_html" type="html">
+                <div>Dear <b>${object.name}</b>,
+                    <br/>
+                    <h3>Thanks for signing up for our newsletter!</h3>
+                    <p>Regards,</p>
+                </div>
+            </field>
+        </record>
+    </data>
+
+    <data>
+        <record id="automated_action" model="base.automation">
+            <field name="name">Send Welcome Mail to new subscribers on Newsletter</field>
+            <field name="model_id" ref="mass_mailing.model_mail_mass_mailing_contact"/>
+            <field name="state">email</field>
+            <field name="trigger">on_create</field>
```

Comment:
This won't take into account if the user is added by a backend odoo user or if he subscribes himself. Is that what we want? :thinking:

Reply:
I don't think we are able to differentiate both.

Reply:
Now I can not remember any email marketing software that does not send an automated welcome email for new subscribers to a list regardless of the source (import, create, add, forms, ...). It's basic functionality, I suppose that if an Odoo user wants that action to exclude x subscribers they'll have to modify the filter of the action or create a customized one.

Reply:
We can differentiate if the mail sending is triggered via the subscription controller instead of from an automated action.

I think current approach can lead to surprises...